bot/cogs/commands.py

```diff
@@ -547,16 +615,30 @@ async def set_main_guild(self, ctx, guild_id: int = None):
     @xp.command(name="give")
     async def xp_give(self, ctx, target: commands.UserConverter, amount: float):
         profile = levelling.UserProfile(target.id, ctx.guild, self.bot)
-        await profile.give_xp(amount)
-        await self.bot.reply_to_msg(ctx.message, f"Gave {amount} xp to {target.name}. "
-                                                 f"They are now rank {profile.rank} ({profile.xp_count} xp)")
+        if not await profile.give_xp(amount):
```

Comment:
This supposes that the only way giving fails is if the amount was negative. Please use a direct check to not rely on such volatile behaviour

Reply:
It is *currently*, I know. It may change and be changed by someone that doesn't know how it's used everywhere

Reply:
a check on amount